crates/weaver_semconv/src/v2/attribute.rs

```diff
@@ -0,0 +1,170 @@
+        default
+    )]
+    deprecated: Option<Deprecated>,
+    /// Specifies the prefix of the attribute.
+    /// If this parameter is set, the resolved id of the referenced attribute will
+    /// have group prefix added to it.
+    /// It defaults to false.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "<&bool>::not")]
+    prefix: bool,
+    /// Annotations for the attribute.
+    annotations: Option<BTreeMap<String, YamlValue>>,
+}
+
+impl AttributeRef {
+    /// Converts a v2 refinement into a v1 AttributeSpec.
+    #[must_use]
+    pub fn into_v1_attribute(self) -> AttributeSpec {
+        AttributeSpec::Ref {
+            r#ref: self.r#ref,
+            brief: self.brief,
+            examples: self.examples,
+            tag: self.tag,
+            requirement_level: self.requirement_level,
+            sampling_relevant: self.sampling_relevant,
+            note: self.note,
+            stability: self.stability,
+            deprecated: self.deprecated,
+            prefix: self.prefix,
+            annotations: self.annotations,
+            role: None,
+        }
+    }
+    /// Converts a v2 refinement into a v1 AttributeSpec.
+    #[must_use]
+    pub fn into_v1_attribute_with_role(self, role: AttributeRole) -> AttributeSpec {
+        AttributeSpec::Ref {
+            r#ref: self.r#ref,
+            brief: self.brief,
+            examples: self.examples,
+            tag: self.tag,
+            requirement_level: self.requirement_level,
+            sampling_relevant: self.sampling_relevant,
+            note: self.note,
+            stability: self.stability,
+            deprecated: self.deprecated,
+            prefix: self.prefix,
+            annotations: self.annotations,
+            role: Some(role),
+        }
+    }
+}
+
+/// The definition of an Attribute.
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "snake_case")]
+pub struct AttributeDef {
+    /// String that uniquely identifies the attribute.
+    pub key: String,
+    /// Either a string literal denoting the type as a primitive or an
+    /// array type, a template type or an enum definition.
+    pub r#type: AttributeType,
+    /// Sequence of example values for the attribute or single example
+    /// value. They are required only for string and string array
+    /// attributes. Example values must be of the same type of the
+    /// attribute. If only a single example is provided, it can directly
+    /// be reported without encapsulating it into a sequence/dictionary.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub examples: Option<Examples>,
+    /// Associates a tag ("sub-group") to the attribute. It carries no
+    /// particular semantic meaning but can be used e.g. for filtering
+    /// in the markdown generator.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub tag: Option<String>,
```

Comment:
Based on the previous comment in `AttributeRef`, we also should remove the tag field into the `AttributeDef` struct.

Reply:
removed